js-envs/src/main/scala/org/scalajs/jsenv/nodejs/JSDOMNodeJSEnv.scala

```diff
@@ -27,79 +27,95 @@ class JSDOMNodeJSEnv(
+  protected class AsyncDOMNodeRunner(files: Seq[VirtualJSFile])
+      extends AsyncExtRunner(files) with AbstractDOMNodeRunner
 
-  protected class ComDOMNodeRunner(libs: Seq[VirtualJSFile], code: VirtualJSFile)
-      extends AsyncDOMNodeRunner(libs, code) with NodeComJSRunner
+  protected class ComDOMNodeRunner(files: Seq[VirtualJSFile])
+      extends AsyncDOMNodeRunner(files) with NodeComJSRunner
 
   protected trait AbstractDOMNodeRunner extends AbstractNodeRunner {
 
     protected def codeWithJSDOMContext(): Seq[VirtualJSFile] = {
-      val scriptsJSPaths = getLibJSFiles().map {
+      val scriptsPaths = getScriptsJSFiles().map {
         case file: FileVirtualFile => file.path
         case file                  => libCache.materialize(file).getAbsolutePath
       }
-      val scriptsStringPath = scriptsJSPaths.map('"' + escapeJS(_) + '"')
+      val scriptsURIs =
+        scriptsPaths.map(path => new java.io.File(path).toURI.toASCIIString)
+      val scriptsURIsAsJSStrings = scriptsURIs.map('"' + escapeJS(_) + '"')
       val jsDOMCode = {
         s"""
            |(function () {
            |  const jsdom = require("jsdom");
-           |  var windowKeys = [];
+           |
+           |  var virtualConsole = jsdom.createVirtualConsole()
+           |    .sendTo(console, { omitJsdomErrors: true });
+           |  virtualConsole.on("jsdomError", function (error) {
+           |    /* This inelegant if + console.error is the only way I found
+           |     * to make sure the stack trace of the original error is
+           |     * printed out.
+           |     */
+           |    if (error.detail && error.detail.stack)
+           |      console.error(error.detail.stack);
+           |
+           |    // Throw the error anew to make sure the whole execution fails
+           |    throw error;
+           |  });
            |
            |  jsdom.env({
            |    html: "",
-           |    virtualConsole: jsdom.createVirtualConsole().sendTo(console),
+           |    url: "http://localhost/",
+           |    virtualConsole: virtualConsole,
            |    created: function (error, window) {
            |      if (error == null) {
            |        window["__ScalaJSEnv"] = __ScalaJSEnv;
            |        window["scalajsCom"] = global.scalajsCom;
-           |        windowKeys = Object.keys(window);
            |      } else {
-           |        console.log(error);
+           |        throw error;
            |      }
            |    },
-           |    scripts: [${scriptsStringPath.mkString(", ")}],
-           |    onload: function (window) {
-           |      jsdom.changeURL(window, "http://localhost");
-           |      for (var k in window) {
-           |        if (windowKeys.indexOf(k) == -1)
-           |          global[k] = window[k];
-           |      }
-           |
-           |      ${code.content}
-           |    }
+           |    scripts: [${scriptsURIsAsJSStrings.mkString(", ")}]
```

Comment:
The fixes to the error handling here seem unrelated to the other changes. Make a separate commit? Also, is there a way to add a test for this?